src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -1222,6 +1220,41 @@ partial class B : A
             Assert.Equal(SpecialType.System_Delegate, typeInfo.ConvertedType!.SpecialType);
         }
 
+        [Fact]
+        public void MethodGroup_BaseAndDerivedTypes_1()
+        {
+            var source = """
+new B().M();
+
+partial class B
+{
+    public void M()
+    {
+        System.Delegate d = F;
+        System.Console.Write(d.GetDelegateTypeName());
```

Comment:
>d.GetDelegateTypeName() [](http://example.com/codeflow?start=29&length=23)

I think it will be useful to invoke the delegate and observe which method is invoked.